java/src/org/openqa/selenium/grid/node/local/LocalNodeFactory.java

```diff
@@ -71,14 +71,15 @@ public static Node create(Config config) {
             .enableCdp(nodeOptions.isCdpEnabled())
             .enableBiDi(nodeOptions.isBiDiEnabled())
             .enableManagedDownloads(nodeOptions.isManagedDownloadsEnabled())
-            .heartbeatPeriod(nodeOptions.getHeartbeatPeriod());
+            .heartbeatPeriod(nodeOptions.getHeartbeatPeriod())
+            .slotMatcher(nodeOptions.getSlotMatcher());
 
     List<DriverService.Builder<?, ?>> builders = new ArrayList<>();
     ServiceLoader.load(DriverService.Builder.class).forEach(builders::add);
 
     nodeOptions
         .getSessionFactories(
-            caps -> createSessionFactory(tracer, clientFactory, sessionTimeout, builders, caps))
+            caps -> createSessionFactory(tracer, clientFactory, sessionTimeout, builders, caps, nodeOptions.getSlotMatcher()))
```

Comment:
I believe this is not needed anymore.

Reply:
I think this is still needed because `createSessionFactory` used the DefaultSlotMatcher